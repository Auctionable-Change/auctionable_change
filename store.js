import React, { createContext, useContext, useReducer } from "react";

const StoreContext = createContext();

const initialState = {
  currentListing: {
    id: null,
    title: "",
    description: "",
    price: null,
    donor: "",
    donor_email: "",
    status: "",
    category: "",
    charity: "",
    charity_url: "",
    charity_score: "",
    charity_score_image: "",
    image: "",
    bids: [],
  },
  buyerDetails: {
    item_id: null,
    bidder_name: "",
    bidder_email: "",
    amount: null,
    street_address: "",
    city: "",
    state: "",
    zip_code: "",
    receipt: "",
  },
  listingToPost: {},
};

const reducer = (state, action) => {
  switch (action.type) {
    case "ADD_CURRENT_LISTING":
      return {
        ...state,
        currentListing: action.currentListing,
      };
    case "ADD_SELECTED_CHARITY":
      return {
        ...state,
        listingToPost: {
          ...state.listingToPost,
          charity: action.selectedCharity.name,
          charity_url: action.selectedCharity.url,
          charity_score: action.selectedCharity.rating,
          charity_score_image: action.selectedCharity.rating_image,
        },
      };
    case "ADD_BUYER_DETAILS":
      return {
        ...state,
        buyerDetails: { ...state.buyerDetails, ...action.buyerDetails },
      };
    case "ADD_TO_LISTING":
      return {
        ...state,
        listingToPost: { ...state.listingToPost, ...action.listingToPost },
      };
    case "POST_BID":
      return {
        ...state,
        currentBid: action.bid
      };
      case "SET_USER":
      return {
        ...state,
        loginDetails: action.loginDetails,
        userInfo: action.userInfo,
        registrationDetails: action.registration
      }

    default:
      throw new Error(`Unhandled action type: ${action.type}`);
  }
};

export const StoreProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);

  return (
    <StoreContext.Provider value={{ state, dispatch }}>
      {children}
    </StoreContext.Provider>
  );
};

export const useStore = () => useContext(StoreContext);
