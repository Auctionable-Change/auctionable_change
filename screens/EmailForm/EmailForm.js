import React from "react";
import { Text, StyleSheet } from "react-native";
import { SafeAreaView } from "react-native-safe-area-context";
import { KeyboardAwareScrollView } from "react-native-keyboard-aware-scroll-view";
import * as MailComposer from "expo-mail-composer";
import { useStore } from "../../store";
import BuyerForm from "../../components/BuyerForm/BuyerForm";

const EmailForm = ({ navigation }) => {
  const { state } = useStore();

  const sendEmail = (buyerObj) => {
    MailComposer.composeAsync({
      recipients: [`${state.currentListing.donor_email}`],
      subject: "Time to ship your item from Auctionable Change",
      body: `Hello ${state.currentListing.donor}, 

      I just made my donation to ${state.currentListing.charity}. Excited to have made A.change with you. Find my info below for shipping.

    Thank you,
    ${buyerObj.bidder_name}

    Email: ${buyerObj.bidder_email}
    Address: ${buyerObj.street_address}, ${buyerObj.city}, ${buyerObj.state} ${buyerObj.zip_code}
    Donation Receipt URL: ${state.buyerDetails.receipt} 
      `,
    });
    navigation.navigate("Thank You");
  };

  return (
    <SafeAreaView style={{ flex: 1, backgroundColor: "#FFFFFF" }}>
      <KeyboardAwareScrollView>
        <Text style={{ fontFamily: "quicksand", fontSize: 20, margin: 5 }}>
          Please enter your contact and shipping information to email the
          seller.
        </Text>
        <BuyerForm navigation={navigation} sendEmail={sendEmail} />
      </KeyboardAwareScrollView>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({});

export default EmailForm;
