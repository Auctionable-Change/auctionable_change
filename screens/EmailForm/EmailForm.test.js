import React from 'react';
import { render, fireEvent, waitFor, NativeTestEvent } from 'react-native-testing-library';
import EmailForm from './EmailForm';
import { StoreProvider } from '../../store'
import {expect, it} from '@jest/globals';
import renderer from "react-test-renderer";


describe('EmailForm', () => {
  it('should render', () => {
    const tree = renderer.create(
      <StoreProvider>
        <EmailForm />
      </StoreProvider>).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('should display passed in information', () => {
    const { getByText } = render(
      <StoreProvider>
        <EmailForm />
      </StoreProvider>
    )
    const message = getByText("3. Enter your contact and shipping information to email the seller.")
    expect(message).toBeTruthy()
  })
})