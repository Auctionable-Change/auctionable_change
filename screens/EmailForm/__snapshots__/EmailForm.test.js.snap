// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailForm should render 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <Text
        style={
          Object {
<<<<<<< HEAD
            "fontFamily": "quicksand",
            "fontSize": 20,
            "margin": 5,
=======
            "alignSelf": "center",
            "backgroundColor": "#FFFFFF",
            "fontFamily": "quicksand-bold",
            "fontSize": 20,
            "textAlign": "center",
            "width": "85%",
>>>>>>> b6751359646ff63db57637c857331c158a511815
          }
        }
      >
        Please enter your contact and shipping information to email the seller.
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "flex": 1,
            "height": 1334,
          }
        }
      >
        <RCTSafeAreaView
          emulateUnlessSupported={true}
          style={
            Object {
              "backgroundColor": "transparent",
              "flex": 1,
            }
          }
        >
          <RCTScrollView
            automaticallyAdjustContentInsets={false}
            contentContainerStyle={
              Array [
                Object {
                  "padding": undefined,
                },
                undefined,
              ]
            }
            contentInset={
              Object {
                "bottom": 0,
              }
            }
            enableAutomaticScroll={true}
            enableOnAndroid={false}
            enableResetScrollToCoords={true}
            extraHeight={75}
            extraScrollHeight={0}
            getScrollResponder={[Function]}
            handleOnScroll={[Function]}
            keyboardDismissMode="interactive"
            keyboardOpeningTime={250}
            keyboardShouldPersistTaps="handled"
            keyboardSpace={0}
            onScroll={[Function]}
            resetKeyboardSpace={[Function]}
            resetScrollToCoords={
              Object {
                "x": 0,
                "y": 0,
              }
            }
            scrollEventThrottle={1}
            scrollForExtraHeightOnAndroid={[Function]}
            scrollIntoView={[Function]}
            scrollToEnd={[Function]}
            scrollToFocusedInput={[Function]}
            scrollToPosition={[Function]}
            showsVerticalScrollIndicator={true}
            style={
              Object {
                "backgroundColor": "transparent",
                "flex": 1,
              }
            }
            update={[Function]}
            viewIsInsideTabBar={false}
          >
            <View>
              <View
                style={Object {}}
              >
                <View
                  accessible={true}
                  focusable={false}
                  forwardedRef={[Function]}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "borderBottomWidth": 1,
                      "borderColor": "#D9D5DC",
                      "borderLeftWidth": 0,
                      "borderRadius": undefined,
                      "borderRightWidth": 0,
                      "borderTopWidth": 0,
                      "borderWidth": undefined,
                      "flexDirection": "row",
                      "marginLeft": 15,
                      "marginTop": 15,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "left": 0,
                        "opacity": 1,
                        "paddingBottom": undefined,
                        "paddingTop": undefined,
                        "position": "absolute",
                        "right": 0,
                        "top": 18,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#575757",
                            "fontSize": 17,
                            "left": 0,
                            "paddingRight": 5,
                            "paddingTop": 5,
                            "top": 6,
                          },
                          Object {
                            "fontFamily": "quicksand",
                            "fontSize": 15,
                          },
                        ]
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "fontSize": 17,
                            },
                            Object {
                              "fontFamily": "quicksand",
                              "fontSize": 15,
                            },
                          ]
                        }
                      >
                        Your Name
                      </Text>
                    </Text>
                  </View>
                  <TextInput
                    allowFontScaling={true}
                    editable={true}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder={null}
                    placeholderTextColor="#575757"
                    rejectResponderTermination={true}
                    style={
                      Object {
                        "color": "#000",
                        "flex": 1,
                        "fontSize": 17,
                        "height": 50,
                        "marginLeft": null,
                        "paddingBottom": 7,
                        "paddingLeft": null,
                        "paddingRight": 5,
                        "paddingTop": 3,
                        "top": 10,
                      }
                    }
                    underlineColorAndroid="rgba(0,0,0,0)"
                    value=""
                  />
                </View>
                <View
                  accessible={true}
                  focusable={false}
                  forwardedRef={[Function]}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "borderBottomWidth": 1,
                      "borderColor": "#D9D5DC",
                      "borderLeftWidth": 0,
                      "borderRadius": undefined,
                      "borderRightWidth": 0,
                      "borderTopWidth": 0,
                      "borderWidth": undefined,
                      "flexDirection": "row",
                      "marginLeft": 15,
                      "marginTop": 15,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "left": 0,
                        "opacity": 1,
                        "paddingBottom": undefined,
                        "paddingTop": undefined,
                        "position": "absolute",
                        "right": 0,
                        "top": 18,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#575757",
                            "fontSize": 17,
                            "left": 0,
                            "paddingRight": 5,
                            "paddingTop": 5,
                            "top": 6,
                          },
                          Object {
                            "fontFamily": "quicksand",
                            "fontSize": 15,
                          },
                        ]
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "fontSize": 17,
                            },
                            Object {
                              "fontFamily": "quicksand",
                              "fontSize": 15,
                            },
                          ]
                        }
                      >
                        Your Email
                      </Text>
                    </Text>
                  </View>
                  <TextInput
                    allowFontScaling={true}
                    editable={true}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder={null}
                    placeholderTextColor="#575757"
                    rejectResponderTermination={true}
                    style={
                      Object {
                        "color": "#000",
                        "flex": 1,
                        "fontSize": 17,
                        "height": 50,
                        "marginLeft": null,
                        "paddingBottom": 7,
                        "paddingLeft": null,
                        "paddingRight": 5,
                        "paddingTop": 3,
                        "top": 10,
                      }
                    }
                    underlineColorAndroid="rgba(0,0,0,0)"
                    value=""
                  />
                </View>
                <View
                  accessible={true}
                  focusable={false}
                  forwardedRef={[Function]}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "borderBottomWidth": 1,
                      "borderColor": "#D9D5DC",
                      "borderLeftWidth": 0,
                      "borderRadius": undefined,
                      "borderRightWidth": 0,
                      "borderTopWidth": 0,
                      "borderWidth": undefined,
                      "flexDirection": "row",
                      "marginLeft": 15,
                      "marginTop": 15,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "left": 0,
                        "opacity": 1,
                        "paddingBottom": undefined,
                        "paddingTop": undefined,
                        "position": "absolute",
                        "right": 0,
                        "top": 18,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#575757",
                            "fontSize": 17,
                            "left": 0,
                            "paddingRight": 5,
                            "paddingTop": 5,
                            "top": 6,
                          },
                          Object {
                            "fontFamily": "quicksand",
                            "fontSize": 15,
                          },
                        ]
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "fontSize": 17,
                            },
                            Object {
                              "fontFamily": "quicksand",
                              "fontSize": 15,
                            },
                          ]
                        }
                      >
                        Street Address
                      </Text>
                    </Text>
                  </View>
                  <TextInput
                    allowFontScaling={true}
                    editable={true}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder={null}
                    placeholderTextColor="#575757"
                    rejectResponderTermination={true}
                    style={
                      Object {
                        "color": "#000",
                        "flex": 1,
                        "fontSize": 17,
                        "height": 50,
                        "marginLeft": null,
                        "paddingBottom": 7,
                        "paddingLeft": null,
                        "paddingRight": 5,
                        "paddingTop": 3,
                        "top": 10,
                      }
                    }
                    underlineColorAndroid="rgba(0,0,0,0)"
                    value=""
                  />
                </View>
                <View
                  accessible={true}
                  focusable={false}
                  forwardedRef={[Function]}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "borderBottomWidth": 1,
                      "borderColor": "#D9D5DC",
                      "borderLeftWidth": 0,
                      "borderRadius": undefined,
                      "borderRightWidth": 0,
                      "borderTopWidth": 0,
                      "borderWidth": undefined,
                      "flexDirection": "row",
                      "marginLeft": 15,
                      "marginTop": 15,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "left": 0,
                        "opacity": 1,
                        "paddingBottom": undefined,
                        "paddingTop": undefined,
                        "position": "absolute",
                        "right": 0,
                        "top": 18,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#575757",
                            "fontSize": 17,
                            "left": 0,
                            "paddingRight": 5,
                            "paddingTop": 5,
                            "top": 6,
                          },
                          Object {
                            "fontFamily": "quicksand",
                            "fontSize": 15,
                          },
                        ]
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "fontSize": 17,
                            },
                            Object {
                              "fontFamily": "quicksand",
                              "fontSize": 15,
                            },
                          ]
                        }
                      >
                        City
                      </Text>
                    </Text>
                  </View>
                  <TextInput
                    allowFontScaling={true}
                    editable={true}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder={null}
                    placeholderTextColor="#575757"
                    rejectResponderTermination={true}
                    style={
                      Object {
                        "color": "#000",
                        "flex": 1,
                        "fontSize": 17,
                        "height": 50,
                        "marginLeft": null,
                        "paddingBottom": 7,
                        "paddingLeft": null,
                        "paddingRight": 5,
                        "paddingTop": 3,
                        "top": 10,
                      }
                    }
                    underlineColorAndroid="rgba(0,0,0,0)"
                    value=""
                  />
                </View>
                <View
                  accessible={true}
                  focusable={false}
                  forwardedRef={[Function]}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "borderBottomWidth": 1,
                      "borderColor": "#D9D5DC",
                      "borderLeftWidth": 0,
                      "borderRadius": undefined,
                      "borderRightWidth": 0,
                      "borderTopWidth": 0,
                      "borderWidth": undefined,
                      "flexDirection": "row",
                      "marginLeft": 15,
                      "marginTop": 15,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "left": 0,
                        "opacity": 1,
                        "paddingBottom": undefined,
                        "paddingTop": undefined,
                        "position": "absolute",
                        "right": 0,
                        "top": 18,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#575757",
                            "fontSize": 17,
                            "left": 0,
                            "paddingRight": 5,
                            "paddingTop": 5,
                            "top": 6,
                          },
                          Object {
                            "fontFamily": "quicksand",
                            "fontSize": 15,
                          },
                        ]
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "fontSize": 17,
                            },
                            Object {
                              "fontFamily": "quicksand",
                              "fontSize": 15,
                            },
                          ]
                        }
                      >
                        State
                      </Text>
                    </Text>
                  </View>
                  <TextInput
                    allowFontScaling={true}
                    editable={true}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder={null}
                    placeholderTextColor="#575757"
                    rejectResponderTermination={true}
                    style={
                      Object {
                        "color": "#000",
                        "flex": 1,
                        "fontSize": 17,
                        "height": 50,
                        "marginLeft": null,
                        "paddingBottom": 7,
                        "paddingLeft": null,
                        "paddingRight": 5,
                        "paddingTop": 3,
                        "top": 10,
                      }
                    }
                    underlineColorAndroid="rgba(0,0,0,0)"
                    value=""
                  />
                </View>
                <View
                  accessible={true}
                  focusable={false}
                  forwardedRef={[Function]}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "borderBottomWidth": 1,
                      "borderColor": "#D9D5DC",
                      "borderLeftWidth": 0,
                      "borderRadius": undefined,
                      "borderRightWidth": 0,
                      "borderTopWidth": 0,
                      "borderWidth": undefined,
                      "flexDirection": "row",
                      "marginLeft": 0,
                      "marginTop": 15,
                      "opacity": 1,
                      "paddingLeft": 15,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "left": 15,
                        "opacity": 1,
                        "paddingBottom": undefined,
                        "paddingTop": undefined,
                        "position": "absolute",
                        "right": 0,
                        "top": 18,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#575757",
                            "fontSize": 17,
                            "left": 0,
                            "paddingRight": 5,
                            "paddingTop": 5,
                            "top": 6,
                          },
                          Object {
                            "fontFamily": "quicksand",
                            "fontSize": 15,
                          },
                        ]
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "fontSize": 17,
                            },
                            Object {
                              "fontFamily": "quicksand",
                              "fontSize": 15,
                            },
                          ]
                        }
                      >
                        Zipcode
                      </Text>
                    </Text>
                  </View>
                  <TextInput
                    allowFontScaling={true}
                    editable={true}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder={null}
                    placeholderTextColor="#575757"
                    rejectResponderTermination={true}
                    style={
                      Object {
                        "color": "#000",
                        "flex": 1,
                        "fontSize": 17,
                        "height": 50,
                        "marginLeft": null,
                        "paddingBottom": 7,
                        "paddingLeft": null,
                        "paddingRight": 5,
                        "paddingTop": 3,
                        "top": 10,
                      }
                    }
                    underlineColorAndroid="rgba(0,0,0,0)"
                    value=""
                  />
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  forwardedRef={[Function]}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "backgroundColor": "#2cb833",
                      "borderBottomWidth": null,
                      "borderColor": "#007aff",
                      "borderLeftWidth": null,
                      "borderRadius": 57,
                      "borderRightWidth": null,
                      "borderTopWidth": null,
                      "borderWidth": undefined,
                      "elevation": 2,
                      "flexDirection": "row",
                      "height": 45,
                      "justifyContent": "space-between",
                      "margin": 10,
                      "opacity": 1,
                      "paddingBottom": 6,
                      "paddingTop": 6,
                      "shadowColor": undefined,
                      "shadowOffset": undefined,
                      "shadowOpacity": undefined,
                      "shadowRadius": undefined,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "color": "#fff",
                          "fontFamily": "System",
                          "fontSize": 16.5,
                          "marginLeft": 0,
                          "marginRight": 0,
                          "paddingLeft": 16,
                          "paddingRight": 16,
                        },
                        Object {
                          "fontFamily": "quicksand-bold",
                          "fontSize": 15,
                        },
                      ]
                    }
                  >
                    Complete Purchase
                  </Text>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </RCTSafeAreaView>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
